src/ctia/init.clj

```diff
@@ -96,20 +96,29 @@
 
 (defn init-store-service! []
   (store-factory-cleaner)
-  (doseq [[store-name store-atom] store/stores]
-    (let [store-type (keyword (get-in @p/properties [:ctia :store store-name] "none"))
-          store-properties (merge (get-in @p/properties [:ctia :store store-type :default] {})
-                                  (get-in @p/properties [:ctia :store store-type store-name] {}))
-          builder (get-in store-factories [:builder store-type] (fn default-builder [f p] (f)))
-          factory (get-in store-factories
-                          [store-name store-type]
-                          #(throw (ex-info (format "Could not configure %s store"
-                                                   store-type)
-                                           {:store-name store-name
-                                            :store-type store-type})))]
-      (if (= store-type :none)
-        (reset! store-atom nil)
-        (reset! store-atom (builder factory store-properties))))))
+  (doseq [[entity-key impls] @store/stores]
+    (swap! store/stores assoc entity-key []))
+
+  (doseq [[store-key store-list] @store/stores]
+    (let [store-impls (or (-> (get-in @p/properties [:ctia :store store-key])
+                              (clojure.string/split #",")) [])
+          store-impl-keys (map keyword store-impls)
+          store-properties (map (fn [impl]
+                                  {:properties (merge (get-in @p/properties [:ctia :store impl :default] {})
+                                                      (get-in @p/properties [:ctia :store impl store-key] {}))
+
+                                   :builder (get-in store-factories [:builder impl]
+                                                    (fn default-builder [f p] (f)))
+
+                                   :factory (get-in store-factories [store-key impl]
+                                                    #(throw (ex-info (format "Could not configure %s store" impl)
+                                                                     {:store-key store-key
+                                                                      :store-type (keyword %)})))}) store-impl-keys)
```

Comment:
Try not to leave "dangling" parameters, like 'store-impl-keys here.  They are hard to find, and can be easily missed by the reader.  Whenever you have multi-line expressions like this, close it on one line and start others on the next line.